### PR TITLE
KDE.Dolphin.Release

### PR DESCRIPTION
**Pull request by beedell-roke**
Improvement of the consistency of the version of KDE.Dolphin.Release and all of the other packages that have been created by KDE, and replacement of the URL for the installer with a version that is much more easy to maintain, but shall not cause submission of invalid installers.

This should allow the improvements that have been submitted to http://github.com/microsoft/winget-pkgs/issues/30234 to be merged.

- [X] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [ ] Have you [validated](https://github.com/denelon/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`? 
- [ ] Have you tested your manifest locally with `winget install --manifest <path>`?
- [ ] Does your manifest conform to the [1.0 schema](https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv1.0.md)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/33718)

### CODE REVIEW

**Comment by KaranKad** [comment]
I think adding `.Release` to stable application might be unnecessary because other stable applications don't have it. Only Nightly, Pre-release, Beta type of applications need it.

File: manifests/k/KDE/Dolphin/21.04.3/KDE.Dolphin.installer.yaml [hunk 1]
```diff
@@ -1,16 +1,16 @@
 # Created using wingetcreate 0.4.1.1
 # yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.0.0.schema.json
 
-PackageIdentifier: KDE.Dolphin
-PackageVersion: 21.04.3
+PackageIdentifier: KDE.Dolphin.Release
+PackageVersion: master-1255
```
Comment (ItzLevvie) [inline manifests/k/KDE/Dolphin/21.04.3/KDE.Dolphin.installer.yaml:5]: `PackageVersion` must match Control Panel.

**Review by ItzLevvie — CHANGES_REQUESTED** [review]

File: manifests/k/KDE/Dolphin/21.04.3/KDE.Dolphin.installer.yaml [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
 Scope: machine
 InstallModes:
 - interactive
 - silent
 Installers:
 - Architecture: x64
   InstallerType: nullsoft
-  InstallerUrl: https://binary-factory.kde.org/job/Dolphin_Release_win64/1255/artifact/dolphin-21.04.3-1255-windows-msvc2019_64-cl.exe
+  InstallerUrl: https://binary-factory.kde.org/view/Windows%2064-bit/job/Dolphin_Release_win64/lastSuccessfulBuild/artifact/dolphin-21.04.3-1255-windows-msvc2019_64-cl.exe
```
Comment (ItzLevvie) [inline manifests/k/KDE/Dolphin/21.04.3/KDE.Dolphin.installer.yaml:13]: Why was this changed? `lastSuccessfulBuild` in the URL is only valid for 1 day, compared to having the build number in the URL which would last for about 4 or 5 days

**Review by ItzLevvie — CHANGES_REQUESTED** [review]

File: manifests/k/KDE/Dolphin/21.04.3/KDE.Dolphin.installer.yaml [hunk 3]
```diff
@@ -1,16 +1,16 @@
 # Created using wingetcreate 0.4.1.1
 # yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.0.0.schema.json
 
-PackageIdentifier: KDE.Dolphin
-PackageVersion: 21.04.3
+PackageIdentifier: KDE.Dolphin.Release
```
Comment (ItzLevvie) [inline manifests/k/KDE/Dolphin/21.04.3/KDE.Dolphin.installer.yaml:4]:
```suggestion
PackageIdentifier: KDE.Dolphin
```
This isn't needed here.

**Review by ItzLevvie — CHANGES_REQUESTED** [review]

**Comment by ItzLevvie** [comment]
> Please do observe http://github.com/microsoft/winget-pkgs/pull/33717#issuecomment-959212573. Consequently, is the reversion that you have proposed definitely mandatory?

Yes, it's mandatory. I don't think wasting people's internet bandwidth would be a good idea if someone has a limited internet connection :)

`21.04.3` shows up in Control Panel rather than `master-1255`
`master-1255` is bigger than `21.04.3` which will make the upgrades of this application in an infinite loop.

**Review by ItzLevvie — CHANGES_REQUESTED** [review]
https://github.com/microsoft/winget-pkgs/pull/33731#discussion_r742127538

**Comment by BEEDELLROKEJULIANLOCKHART** [comment]
> http://github.com/microsoft/winget-pkgs/pull/33718#issuecomment-959640978.

How would usage of that system cause infinite replacement?
Also, because the nightly versions do not use the standard release versions, what should they use? http://github.com/microsoft/winget-pkgs/pull/33717#issuecomment-959212573 would be much less consistent if that were to be true.

**Review by ItzLevvie — CHANGES_REQUESTED** [review]
> How would usage of that system cause infinite replacement?

When it comes to upgrading applications, WinGet takes a look at the installed version from Control Panel (Registry) and compares it to the package repository (winget-pkgs), since `master-1255` is bigger than the installed version of `21.04.3`, it's considered as an upgrade loop (or an infinity upgrade) because there's a version mismatch between both versions.

So, therefore `21.04.3` from Control Panel must match the one in the package repository.

> Also, because the nightly versions do not use the standard release versions

Generally, the versions must match what's shown in Control Panel.

> would be much less consistent if that were to be true.

If you want the version to be `master-XXXX` I would recommend to file an issue about it in KDE's issue tracker.

You could also file an issue on their issue tracker, for allowing other people to upgrade from version to version by asking the developer to add the Jenkins build number at the end of the version in Control Panel,  say `21.04.3.1255`.

**Comment by BEEDELLROKEJULIANLOCKHART** [comment]
> http://github.com/microsoft/winget-pkgs/pull/33718#pullrequestreview-796861130.

I have filed that at "http://bugs.kde.org/show_bug.cgi?id=444894". Should we wait until that has been resolved, or should I revert that section of my modification until then?

(I am hopeful that my explanation of the problem was adequately correct, specific, and descriptive. If it was not, please do add to it.)

**Comment by BEEDELLROKEJULIANLOCKHART** [comment]
Should I use https://github.com/microsoft/winget-create/releases for remediation of http://github.com/microsoft/winget-pkgs/pull/33718#issuecomment-959052300?

**Comment by ItzLevvie** [comment]
> Should I use "[http://github.com/microsoft/winget-create/releases](https://github.com/microsoft/winget-create/releases)" for remediation of "[http://github.com/microsoft/winget-pkgs/pull/33718#issuecomment-959052300](https://github.com/microsoft/winget-pkgs/pull/33718#issuecomment-959052300)"?

You're free to use either wingetcreate, or YamlCreate.ps1 in the Tools directory in this repository.

YamlCreate.ps1 is preferred because it allows you to add more metadata to the manifest.

**Comment by vedantmgoyal2009** [comment]
> I have filed the report of potential improvement at “[http://bugs.kde.org/show_bug.cgi?id=444894”](http://bugs.kde.org/show_bug.cgi?id=444894%E2%80%9D). Should we wait until that has been resolved, or should I revert that section of my modification via this pull-request until then?

I think they have deleted the bug.

**Comment by ItzLevvie** [comment]
> I think they have deleted the bug.

It's here: https://bugs.kde.org/show_bug.cgi?id=444894

**Comment by vedantmgoyal2009** [comment]
Yeah, I think it was a redirect issue (HTTP -> HTTPS).

**Comment by BEEDELLROKEJULIANLOCKHART** [comment]
This comment is response to "http://github.com/microsoft/winget-pkgs/pull/33718#issuecomment-979974212":

The problem that you have encountered was caused incorrect identification of the hyperlink by the browser because the parser is not able to understand English. Consequently, the successive mark-of-speech was interpreted as constituent of the hyperlink. Do observe for verification if you are able to which sections of the hyperlink are blue.
